pshtt/pshtt.py

```diff
@@ -526,31 +526,40 @@ def hsts_check(endpoint):
             endpoint.hsts = False
             return
 
-        endpoint.hsts = True
         endpoint.hsts_header = header
 
-        # Set max age to the string after max-age
-        # TODO: make this more resilient to pathological HSTS headers.
-
         # handle multiple HSTS headers, requests comma-separates them
-        first_pass = re.split(r',\s?', header)[0]
-        second_pass = re.sub(r'\'', '', first_pass)
+        headers = [x.strip() for x in header.split(",")]
+        # Multiple HSTS headers does not conform to RFCs 7230-3.2.2 and 6797-6.1
+        if len(headers) > 1:
+            logging.warning("Host is incorrectly returning multiple HSTS headers: {}".format(header))
```

Comment:
We could just fail them here, if multiple headers are not allowed according to the RFC.  Did you choose not to for a technical reason?

Reply:
This was definitely a sweeping changes moment. I thought it more conservative to give credit for the first header received vs failing completely. I am quite open to failing them completely on multiple headers (which appears to be along the lines of what SSL Labs does). However, since browsers do not die on it that may be a harder point to get across.

Reply:
This is a point to discuss with the CyHy team, after explaining the situation to them.  I'd in particular want input from @climber-girl, and @h-m-f-t if he is available.

Reply:
After digging in the RFC more deeply it does specify in [Section 8.1](https://tools.ietf.org/html/rfc6797#section-8.1)

>If a UA receives more than one STS header field in an HTTP response message over secure transport, then the UA MUST process only the first such header field.

So although sending multiple headers is wrong, the RFC does have guidelines for how to handle such cases.

Reply:
I agree with @mcdonnnj on changing the code to not immediately fail for multiple headers and rather take the first header.

Reply:
It seems that the consensus is to generate the warning for multiple headers and take the first one.  I'm OK with that.